common/persistence/data_manager_interfaces.go

```diff
@@ -174,6 +174,46 @@ const (
 	TaskListKindSticky
 )
 
+// HistoryTaskCategory represents various categories of history tasks
+type HistoryTaskCategory struct {
+	categoryType int
+	categoryID   int
+}
+
+func (c *HistoryTaskCategory) Type() int {
+	return c.categoryType
+}
+
+func (c *HistoryTaskCategory) ID() int {
+	return c.categoryID
+}
+
+const (
+	HistoryTaskCategoryTypeImmediate = iota + 1
+	HistoryTaskCategoryTypeScheduled
+)
+
+const (
+	HistoryTaskCategoryIDTransfer    = 1
+	HistoryTaskCategoryIDTimer       = 2
+	HistoryTaskCategoryIDReplication = 3
```

Comment:
category id to category type mapping is 1 to 1. Do we need the category type as a separate enum?

Reply:
Yeah, at some place, I found I need to tell if the task is a scheduled task or immediate task. 
And that requires another map to track which category is scheduled task and which category is immediate task.